nss-day-cohort-48/bangazon-tenntom
# Zero Division Error

The get products request was returning as error because a division action was taking place with zero in the denominator. This fix addresses that issue by adding an NAV to all products that have not been reviewed yet. That NAV can be changed to 0 or to "Not Yet Reviewed" or whatever the client would like to put in there. For this branch currently, it is set to NAV.  

In the process of working on this, I realized we had a rating model, which is not connected to the product properly, and a productrating model, which was connected to the product and was producing the zerodenomiator error. For that reason, I removed the rating model.

Changes
 1. Changed the bangazon/vmodels/product -- Edited the average_rating property so that the rating only calculates an average 
 2. when at least one non-0 rating has been received. 
 3. Deleted the ratings model.
 4. Deleted the import of the ratings model from the init file. 

To Test:

1. Git fetch --all
2. Pull branch tlc-zero-div
3. From the command line, run ./seed_data.sh
4. start the server python3 manage.py runserver
5. From postman, send a post request to log-in to get an authorization key. (I think just sending this is enough)
6. From postman, request products . http://localhost:8000/products
7. Check that product 1 has an average rating of 4.5
8. Check that product 2 has an average rating of "NAV"
9. Check that all others have average rating of "NAV" except 50, which has average rating of 3.25.
10. Open this up in MS Code and check to make sure no rating model is present.
11. Confirm in MS Code that the product rating model is there
12. Confirm that the product model has a logical sequence of orders in its average_rating property.

FIXES ticket 15
